File: Java/java2/Java2/src/Student.java

public class Student 
{
	private int number =10;//생성시 10
	static int sample=10;
	
	//static 초기화 블럭
	static
	{//static 맴버를 초기화 하는데 복잡한 연산을 사용해서 초기화를 해야될 경우
		sample =10*20-10*200;
	}
	//인스턴스 초기화 블럭
	{//인스턴스가 호출 될때 마다 실행되고 생성자가 실행됨====그냥 생성자에 넣으면됨
		
	}
	//생성자
	public Student()
	{
		this(0);
	}
	public Student(int number)
	{
		this.number=number;
	}
}

File: WinForm/Test/Test/Student.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Test
{
    enum SubjectType
    {
        NONE,
        COM,
        IT,
        GAME,
        ETC
    }

    class Student
    {
        #region 필드&프로퍼티
        public readonly int number;
        public string name { get; private set; }
        public SubjectType stype { get; set; }
        private int grade;
        public int Grade
        {
            get { return grade; }
            set
            {
                if (value < 0 || value > 4)
                    throw new Exception("해당 학년은 존재하지 않습니다");
                grade = value;
            }
        }
        #endregion

        #region 생성자
        public Student()
        {

        }
        public Student(int number, string name, SubjectType stype, int grade)//회원가입을 진행하는 경우
        {
            this.number = number;
            this.name = name;
            this.stype = stype;
            this.Grade = grade;
        }
        #endregion

        #region 기능 메서드
        public override string ToString()
        {
            return String.Format("{0},{1},{2},{3}", number, name, stype, grade);
        }
        #endregion

    }
}
